From d11ed6ff267c24c6adfa7acf5e2cda63f4df5edf Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Fri, 13 Nov 2015 19:02:10 -0500
Subject: [PATCH] testcases/ddt/Makefile: filter out testcases for k2g

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 testcases/ddt/Makefile | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/testcases/ddt/Makefile b/testcases/ddt/Makefile
index beed4f2..d3e95e7 100644
--- a/testcases/ddt/Makefile
+++ b/testcases/ddt/Makefile
@@ -339,6 +339,17 @@ FILTER_OUT_DIRS                 += ipc_test_suite
 FILTER_OUT_DIRS                 += gpio_test_suite
 endif
 
+ifeq ($(PLATFORM),k2g-evm)
+FILTER_OUT_DIRS                 += edma_test_suite
+FILTER_OUT_DIRS                 += fbdev_display_test_suite
+FILTER_OUT_DIRS                 += mc_test_suite
+FILTER_OUT_DIRS                 += v4l2_capture_test_suite
+FILTER_OUT_DIRS                 += v4l2_display_test_suite
+FILTER_OUT_DIRS                 += v4l2_display_davinci_test_suite
+FILTER_OUT_DIRS                 += ipc_test_suite
+FILTER_OUT_DIRS                 += gpio_test_suite
+endif
+
 ifeq ($(PLATFORM),dragonboard410c)
 FILTER_OUT_DIRS                 += v4l2_capture_test_suite
 FILTER_OUT_DIRS                 += v4l2_display_test_suite
-- 
2.2.0

