From a7b8b39581a195e5f7d5bd4185d59ad146d54bce Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Mon, 6 Apr 2020 18:15:12 -0400
Subject: [PATCH] testcases/kernel/syscalls: disable stime due to glibc-2.31
 deprecation

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 testcases/kernel/syscalls/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/testcases/kernel/syscalls/Makefile b/testcases/kernel/syscalls/Makefile
index 45a00cf..d7b1d3b 100644
--- a/testcases/kernel/syscalls/Makefile
+++ b/testcases/kernel/syscalls/Makefile
@@ -45,4 +45,6 @@ FILTER_OUT_DIRS	+= \
 	setdomainname sethostname symlinkat ulimit ustat vfork vhangup vmsplice
 endif
 
+FILTER_OUT_DIRS	+= stime
+
 include $(top_srcdir)/include/mk/generic_trunk_target.mk
-- 
2.7.4

