--- strongswan-5.0.0/src/strongswan.conf	2010-10-22 10:33:30.000000000 -0400
+++ strongswan-5.0.0.libsecstore/src/strongswan.conf	2014-04-03 18:31:17.228569896 -0400
@@ -31,4 +31,13 @@
 
 	#  set to no, the DH exponent size is optimized
 	#  dh_exponent_ansi_x9_42 = no
+  plugins {
+    pkcs11 {
+      modules {
+        secstore {
+          path = /usr/lib/softhsm/libsecstore.so.1
+        }
+      }
+    }
+  }
 }
