From 394eacf80efaacc719e69179c7c709016d21b707 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Mon, 15 Dec 2014 15:46:08 -0500
Subject: [PATCH] Makefile.build: with Qt5 use qmake available in PATH

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 Makefile.build |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.build b/Makefile.build
index a0ceaad..e6cb010 100644
--- a/Makefile.build
+++ b/Makefile.build
@@ -8,11 +8,11 @@ all: release
 .PHONY: qmake
 qmake : refresh_screen.pro
 	@. ${ENV_SETUP}; \
-	qmake2 CONFIG+=release refresh_screen.pro
+	qmake CONFIG+=release refresh_screen.pro
 
 qmake_debug : refresh_screen.pro
 	@. ${ENV_SETUP}; \
-	qmake2 CONFIG+=debug refresh_screen.pro
+	qmake CONFIG+=debug refresh_screen.pro
 
 debug : qmake_debug
 	@. ${ENV_SETUP}; \
-- 
1.7.9.5

