From 60a8e6c8ea763ed88442f16509cae99d8e43918e Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Mon, 12 May 2014 21:01:57 -0400
Subject: [PATCH] Replace QtGui header with QtWidgets for Qt5

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.cpp b/main.cpp
index c5744d8..5a61904 100644
--- a/main.cpp
+++ b/main.cpp
@@ -36,7 +36,7 @@
  *
 */
 
-#include <QtGui/QApplication>
+#include <QtWidgets/QApplication>
 #include "mainwindow.h"
 
 int main(int argc, char *argv[])
-- 
1.9.2

