diff --git a/scripts/setup.sh b/scripts/setup.sh
index 7bba711..4a98e66 100755
--- a/scripts/setup.sh
+++ b/scripts/setup.sh
@@ -41,6 +41,12 @@ MODULES=(
 "ietf-network-instance@2019-01-21.yang"
 "ietf-subscribed-notifications@2019-09-09.yang -e encode-xml -e replay -e subtree -e xpath"
 "ietf-yang-push@2019-09-09.yang -e on-change"
+"iana-if-type@2017-01-19.yang"
+"ieee802-types@2022-10-31.yang"
+"ieee802-dot1q-types@2022-10-31.yang"
+"ieee802-dot1q-bridge@2022-10-31.yang"
+"ieee802-dot1q-sched@2022-08-18.yang"
+"ieee802-dot1q-sched-bridge@2022-08-18.yang"
 )

 CMD_INSTALL=
