From 42c76d70760a8b0d19a4126d3c53578f5f5b3606 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Sun, 21 Dec 2014 14:18:46 -0500
Subject: [PATCH] switch-config.c: include stdbool.h to define bool type

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 switch-config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/switch-config.c b/switch-config.c
index d49cac7..0d1f9f9 100644
--- a/switch-config.c
+++ b/switch-config.c
@@ -19,6 +19,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <stddef.h>
+#include <stdbool.h>
 #include <getopt.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
-- 
2.2.0

