From dbaac14600fd88988aa676634849b137376de397 Mon Sep 17 00:00:00 2001
From: Karthik Ramanan <a0393906@ti.com>
Date: Thu, 15 Jan 2015 12:47:31 +0530
Subject: [PATCH] waylandsink: Removed dependency on dri2

Signed-off-by: Karthik Ramanan <a0393906@ti.com>
---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e3db68f..2741824 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1219,7 +1219,7 @@ AG_GST_CHECK_FEATURE(DIRECTFB, [directfb], dfbvideosink , [
 dnl **** Wayland ****
 translit(dnm, m, l) AM_CONDITIONAL(USE_WAYLAND, true)
 AG_GST_CHECK_FEATURE(WAYLAND, [wayland sink], wayland , [
-  PKG_CHECK_MODULES(WAYLAND, wayland-client >= 1.0.0 dri2 libdrm libdrm_omap, [
+  PKG_CHECK_MODULES(WAYLAND, wayland-client >= 1.0.0 libdrm libdrm_omap, [
      AC_SUBST(DRM_CFLAGS)
      AC_SUBST(DRM_LIBS)
      HAVE_WAYLAND="yes" ], [ HAVE_WAYLAND="no"
-- 
1.7.9.5

