From 4f5d63b026f225f6499e0a401d9b2769e8ef0a80 Mon Sep 17 00:00:00 2001
From: Djordje Senicic <x0157990@ti.com>
Date: Wed, 12 Jun 2019 13:12:09 -0400
Subject: [PATCH] Avoid stringop-overflow warning resulting in compilation
 error

Upstream-Status: Inappropriate [TI only test code]

Signed-off-by: Djordje Senicic <x0157990@ti.com>
---
 src/armnn/LayerSupport.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/armnn/LayerSupport.cpp b/src/armnn/LayerSupport.cpp
index 12b3c403..09c973e5 100644
--- a/src/armnn/LayerSupport.cpp
+++ b/src/armnn/LayerSupport.cpp
@@ -25,6 +25,8 @@ void CopyErrorMessage(char* truncatedString, const char* fullString, size_t maxL
     if(truncatedString != nullptr)
     {
         size_t copyLength = std::min(maxLength, strlen(fullString));
+        if(copyLength > 255) copyLength = 255; //Put source length independent upper bound
+       
         std::strncpy(truncatedString, fullString, copyLength);
         // Ensure null-terminated string.
         truncatedString[copyLength] = '\0';
-- 
2.17.1

