diff -uNr icu-orig/config/mh-linux icu/config/mh-linux
--- icu-orig/config/mh-linux	2013-05-23 16:02:40.000000000 -0400
+++ icu/config/mh-linux	2013-09-30 00:19:13.772269345 -0400
@@ -21,7 +21,7 @@
 LD_RPATH_PRE = -Wl,-rpath,
 
 ## These are the library specific LDFLAGS
-LDFLAGSICUDT=-nodefaultlibs -nostdlib
+#LDFLAGSICUDT=-nodefaultlibs -nostdlib
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
