From 32bd788608034f5a49d9cd4a0163190ef666d9a2 Mon Sep 17 00:00:00 2001
From: Djordje Senicic <d-senicic1@ti.com>
Date: Mon, 14 Aug 2017 10:26:39 -0400
Subject: [PATCH] Fix paths for Yocto build and add RTOS build var

Signed-off-by: Djordje Senicic <d-senicic1@ti.com>
---
 Makefile            | 27 +++++++++++++++++++--------
 psdk_tools_path.mak | 32 ++++++++++++++------------------
 2 files changed, 33 insertions(+), 26 deletions(-)

diff --git a/Makefile b/Makefile
index 0832e07..1be5b0d 100755
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 
 # Valid values are: vsdk psdk
-BUILD_SDK=vsdk
+BUILD_SDK=psdk
 
 include $(BUILD_SDK)_tools_path.mak
 
@@ -8,11 +8,12 @@ BUILD_EMULATION_MODE?=no
 BUILD_TARGET_MODE?=yes
 
 BUILD_CONFORMANCE_TEST?=yes
-BUILD_IVISION_KERNELS?=yes
+BUILD_IVISION_KERNELS?=no
 BUILD_BAM?=yes
 BUILD_TUTORIAL?=yes
 BUILD_LINUX_A15?=yes
-BUILD_EVE?=yes
+BUILD_RTOS?=no
+BUILD_EVE?=no
 
 PROFILE?=all
 
@@ -43,12 +44,17 @@ TARGET_COMBOS :=
 
 ifeq ($(BUILD_TARGET_MODE),yes)
   ifeq ($(PROFILE), $(filter $(PROFILE), debug all))
-	TARGET_COMBOS += TDAX:SYSBIOS:M4:1:debug:TIARMCGT
-	TARGET_COMBOS += TDAX:SYSBIOS:C66:1:debug:CGT6X
+	ifeq ($(BUILD_RTOS),yes)
+	  TARGET_COMBOS += TDAX:SYSBIOS:M4:1:debug:TIARMCGT
+	  TARGET_COMBOS += TDAX:SYSBIOS:C66:1:debug:CGT6X
+	endif
+
 	ifeq ($(BUILD_EVE),yes)
 	TARGET_COMBOS += TDAX:SYSBIOS:EVE:1:debug:ARP32CGT
 	endif
+	ifeq ($(BUILD_RTOS),yes)
 	TARGET_COMBOS += TDAX:SYSBIOS:A15:1:debug:GCC
+	endif
 	ifneq ($(OS),Windows_NT)
 		ifeq ($(BUILD_LINUX_A15),yes)
 		TARGET_COMBOS += TDAX:LINUX:A15:1:debug:GCC_LINARO
@@ -57,12 +63,17 @@ ifeq ($(BUILD_TARGET_MODE),yes)
   endif
 
   ifeq ($(PROFILE), $(filter $(PROFILE), release all))
-	TARGET_COMBOS += TDAX:SYSBIOS:M4:1:release:TIARMCGT
-	TARGET_COMBOS += TDAX:SYSBIOS:C66:1:release:CGT6X
+	ifeq ($(BUILD_RTOS),yes)
+	  TARGET_COMBOS += TDAX:SYSBIOS:M4:1:release:TIARMCGT
+	  TARGET_COMBOS += TDAX:SYSBIOS:C66:1:release:CGT6X
+	endif
+
 	ifeq ($(BUILD_EVE),yes)
 	TARGET_COMBOS += TDAX:SYSBIOS:EVE:1:release:ARP32CGT
 	endif
-	TARGET_COMBOS += TDAX:SYSBIOS:A15:1:release:GCC
+	ifeq ($(BUILD_RTOS),yes)
+	  TARGET_COMBOS += TDAX:SYSBIOS:A15:1:release:GCC
+	endif
 	ifneq ($(OS),Windows_NT)
 		ifeq ($(BUILD_LINUX_A15),yes)
 		TARGET_COMBOS += TDAX:LINUX:A15:1:release:GCC_LINARO
diff --git a/psdk_tools_path.mak b/psdk_tools_path.mak
index 36394f8..8fd468d 100755
--- a/psdk_tools_path.mak
+++ b/psdk_tools_path.mak
@@ -2,7 +2,6 @@
 TIOVX_PATH = $(abspath .)
 
 PLSDK_PATH := /home/x0157990/ti-processor-sdk-linux-am57xx-evm-04.00.00.04
-PRSDK_PATH := /home/x0157990/ti-processor-sdk-rtos-am57xx-evm-04-00-00-04
 
 CROSS_COMPILE_LINARO := arm-linux-gnueabihf-
 CROSS_COMPILE := arm-none-eabi-
@@ -16,23 +15,20 @@ ifeq ($(BUILD_OS),Windows_NT)
 GCC_WINDOWS_ROOT ?= C:/CodeBlocks/MinGW
 endif
 
-XDC_PATH := $(PRSDK_PATH)/xdctools_3_32_01_22_core/
-TIARMCGT_ROOT := $(PRSDK_PATH)/ti-cgt-arm_16.9.2.LTS
+XDC_PATH := $(XDC_INSTALL_DIR)
+TIARMCGT_ROOT :=  $(TIARMCGTPATH)
+GCC_ROOT_LINARO := $(GCCLINARO)
+XDIAS_PATH := $(XDAISPATH)
+BIOS_PATH := $(SYSBIOSPATH)
+PDK_PATH := $(PDKPATH)
+SDK_PLATFORM_IF_PATH := $(SDKPLATFORMIFPATH)
 
-GCC_ROOT := $(PRSDK_PATH)/gcc-arm-none-eabi-4_9-2015q3
-GCC_ROOT_LINARO := /home/x0157990/linaro/gcc-linaro-6.2.1-2016.11-x86_64_arm-linux-gnueabihf
-CGT6X_ROOT := $(PRSDK_PATH)/ti-cgt-c6000_8.1.3/
+GCC_ROOT ?= $(GCCARMNONEPATH)
+CGT6X_ROOT := $(TI_CGT_INSTALL)
 GCC_LINUX_ROOT ?= /usr/
+VXLIB_PATH ?= $(VXLIBPATH)
+ARP32CGT_ROOT ?= $(EVETOOLSPATH}
+DMAUTILS_PATH ?= $(DSPAPPS_PATH)/dmautils
+ALGFRAMEWORK_PATH ?= $(DSPAPPS_PATH)/algframework
+EDMA3_LLD_PATH := $(EDMA3LLDPATH)
 
-
-VXLIB_PATH ?= $(PRSDK_PATH)/vxlib_c66x_1_1_0_0
-XDIAS_PATH := $(PRSDK_PATH)/xdais_7_24_00_04
-EDMA3_LLD_PATH := $(PRSDK_PATH)/edma3_lld_2_12_04_28/
-BIOS_PATH := $(PRSDK_PATH)/bios_6_46_05_55/
-
-DMAUTILS_PATH ?= $(PRSDK_PATH)/REL.DMAUTILS.00.08.00.02/dmautils
-ALGFRAMEWORK_PATH ?= $(PRSDK_PATH)/REL.ALGFRAMEWORK.02.07.00.00/algframework
-PDK_PATH := $(PRSDK_PATH)/pdk_am57xx_1_0_7
-CMEM_PATH ?= $(PLSDK_INSTALL_PATH)/board-support/extra-drivers/cmem-mod-4.14.00.00+gitAUTOINC+b514a99ac4/
-TARGETFS := $(PLSDK_PATH)/linux-devkit/sysroots/armv7ahf-neon-linux-gnueabi/
-SDK_PLATFORM_IF_PATH ?= /home/x0157990/am57/openvx/ex41_forwardmsg/vx
\ No newline at end of file
-- 
1.9.1

