From 50a533a969a0efd9c4c7713ad571b8f853b1b7e5 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Fri, 11 Sep 2015 12:57:56 -0400
Subject: [PATCH] util/demux: update use of deprecated API

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 util/demux.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/util/demux.c b/util/demux.c
index d739519..e932ac4 100644
--- a/util/demux.c
+++ b/util/demux.c
@@ -44,18 +44,18 @@ int get_esds_offset(const char *filename, struct demux *demux);
 static AVFormatContext *
 open_file(const char *filename)
 {
-	AVFormatContext *afc;
-	int err = av_open_input_file(&afc, filename, NULL, 0, NULL);
+	static AVFormatContext *afc = NULL;
+	int err = avformat_open_input(&afc, filename, NULL, NULL);
 
 	if (!err)
-		err = av_find_stream_info(afc);
+		err = avformat_find_stream_info(afc, NULL);
 
 	if (err < 0) {
 		ERROR("%s: lavf error %d", filename, err);
 		exit(1);
 	}
 
-	dump_format(afc, 0, filename, 0);
+	av_dump_format(afc, 0, filename, 0);
 
 	return afc;
 }
-- 
2.2.0

