From 07eeb01ea39e0fa2f9d30900c07fd0345d32562b Mon Sep 17 00:00:00 2001
From: Eric Ruei <e-ruei1@ti.com>
Date: Thu, 1 Sep 2016 17:49:56 -0400
Subject: [PATCH] WebEngine qquickwebengineview_p.h add include
 <QtGui/qcolor.h>

Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 src/webengine/api/qquickwebengineview_p.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/webengine/api/qquickwebengineview_p.h b/src/webengine/api/qquickwebengineview_p.h
index 98d0bbc..93ef9e4 100644
--- a/src/webengine/api/qquickwebengineview_p.h
+++ b/src/webengine/api/qquickwebengineview_p.h
@@ -51,6 +51,7 @@
 #include <private/qtwebengineglobal_p.h>
 #include "qquickwebenginescript_p.h"
 #include <QQuickItem>
+#include <QtGui/qcolor.h>
 
 QT_BEGIN_NAMESPACE
 
-- 
1.9.1

