diff --git a/Source/WebCore/WebCore.pri b/Source/WebCore/WebCore.pri
index 0189394..00f7090 100644
--- a/Source/WebCore/WebCore.pri
+++ b/Source/WebCore/WebCore.pri
@@ -227,8 +227,10 @@ use?(3D_GRAPHICS) {
 use?(GRAPHICS_SURFACE) {
     mac: LIBS += -framework IOSurface -framework CoreFoundation
     linux-*: {
-        LIBS += -lXcomposite -lXrender
-        CONFIG *= x11
+        xlibAvailable() {
+            LIBS += -lXcomposite -lXrender
+            CONFIG *= x11
+        }
     }
 }
 
