From ae986b12f20b88de5f949062516fff4d8f887645 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Mon, 1 May 2017 16:33:33 -0400
Subject: [PATCH] uvc-gadget: don't hardcode uvc.h path

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 uvc-gadget.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uvc-gadget.c b/uvc-gadget.c
index 4d59ab8..cd9e916 100644
--- a/uvc-gadget.c
+++ b/uvc-gadget.c
@@ -36,7 +36,7 @@
 #include <linux/usb/video.h>
 #include <linux/videodev2.h>
 
-#include "../drivers/usb/gadget/uvc.h"
+#include "uvc.h"
 
 #define clamp(val, min, max) ({                 \
         typeof(val) __val = (val);              \
-- 
2.7.4

