From 67f8fdee6e979173c7c6793593147030df6aaa40 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Tue, 8 Sep 2015 17:19:57 -0400
Subject: [PATCH] eng_cryptodev: Add SHA224 initialization to cryptodev engine.

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 crypto/engine/eng_cryptodev.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/crypto/engine/eng_cryptodev.c b/crypto/engine/eng_cryptodev.c
index b7bff89..3cd4bcf 100644
--- a/crypto/engine/eng_cryptodev.c
+++ b/crypto/engine/eng_cryptodev.c
@@ -216,6 +216,9 @@ static struct {
         CRYPTO_SHA1, NID_sha1, 20
     },
     {
+        CRYPTO_SHA2_224, NID_sha224, 28
+    },
+    {
         CRYPTO_SHA2_256, NID_sha256, 32
     },
     {
-- 
2.2.0

