diff --git a/CMake/modules/FindWaylandProtocols.cmake b/CMake/modules/FindWaylandProtocols.cmake
index 8eb83f2..1e64fa9 100644
--- a/CMake/modules/FindWaylandProtocols.cmake
+++ b/CMake/modules/FindWaylandProtocols.cmake
@@ -22,5 +22,5 @@ find_package_handle_standard_args(WaylandProtocols
 )
 
 set(WAYLAND_PROTOCOLS_FOUND ${WaylandProtocols_FOUND})
-set(WAYLAND_PROTOCOLS_PKGDATADIR ${WaylandProtocols_PKGDATADIR})
+set(WAYLAND_PROTOCOLS_PKGDATADIR $ENV{PKG_CONFIG_SYSROOT_DIR}/${WaylandProtocols_PKGDATADIR})
 set(WAYLAND_PROTOCOLS_VERSION ${WaylandProtocols_VERSION})
