From ecba95bde41a91d72aecf569c853a766a6ffd2aa Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Sun, 26 Mar 2017 13:39:47 -0400
Subject: [PATCH] Makefile: fix library linking issues

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 cpuload-plugins/Makefile.am | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/cpuload-plugins/Makefile.am b/cpuload-plugins/Makefile.am
index 9c0bc1a..3df7b22 100644
--- a/cpuload-plugins/Makefile.am
+++ b/cpuload-plugins/Makefile.am
@@ -6,13 +6,12 @@ mpuload_CFLAGS = \
 mpuload_SOURCES = mpuload.c
 
 rprocinfo_CFLAGS = -O0 -g @DRM_CFLAGS@ @DCE_CFLAGS@
-rprocinfo_LDFLAGS = @DCE_LIBS@ -lpthread
+rprocinfo_LDADD = @DCE_LIBS@ -lpthread
 rprocinfo_SOURCES = rprocinfo.c
 
-pvrscope_CXXFLAGS = -I. -lrt -lm -lpthread -ldl
+pvrscope_CXXFLAGS = -I.
+pvrscope_LDADD = $(top_srcdir)/cpuload-plugins/libPVRScopeDeveloper.a -lrt -lm -lpthread -ldl
 pvrscope_SOURCES = pvrscope.cpp
-LIBS = $(top_srcdir)/cpuload-plugins/libPVRScopeDeveloper.a
-
 
 # Specify any libraries that need to be built here.  Follow the convention
 # libxxx.a otherwise autotools will complain about non-canonical names.
@@ -26,17 +25,15 @@ include_HEADERS=
 ##
 
 read32k_driver_SOURCES= read32k_driver.c
-read32k_driver_LDADD = libread32k.a
+read32k_driver_LDADD = libread32k.a -lm
 read32k_driver_CFLAGS = -I.
-read32k_driver_LDFLAGS= -lm
 
 readproc_SOURCES = readproc.c
-readproc_LDFLAGS = -lm
+readproc_LDADD = -lm
 
 waitforweston_SOURCES = waitforweston.c
-waitforweston_LDADD = libread32k.a
+waitforweston_LDADD = libread32k.a -lm @DRM_LIBS@ -lpthread -lrt @WAYLAND_LIBS@ @UDEV_LIBS@
 waitforweston_CFLAGS = -I. @DRM_CFLAGS@
-waitforweston_LDFLAGS = @DRM_LIBS@ -lpthread -lrt -lm @WAYLAND_LIBS@ @UDEV_LIBS@
 
 ##
 # Libraries
@@ -45,4 +42,3 @@ waitforweston_LDFLAGS = @DRM_LIBS@ -lpthread -lrt -lm @WAYLAND_LIBS@ @UDEV_LIBS@
 libread32k_a_SOURCES = read32k.c
 libread32k_a_CFLAGS = -I.
 include_HEADERS+=./read32k.h
-
-- 
2.7.4

