From f49b3d41e74f953825f234258c1a33a5e3b914fc Mon Sep 17 00:00:00 2001
From: "Franklin S. Cooper Jr" <fcooperjr27@gmail.com>
Date: Fri, 7 Sep 2012 18:36:24 -0500

---
 extras/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/Makefile.in b/extras/Makefile.in
index 56942db..f174e3f 100644
--- a/extras/Makefile.in
+++ b/extras/Makefile.in
@@ -36,7 +36,7 @@ CCOPT =		@V_CCOPT@
 DEFS =		@DEFS@
 INCLS =		-I..
 CFLAGS =	$(CCOPT) $(DEFS) $(INCLS)
-STATICFLAG =	@V_STATICFLAG@
+STATICFLAG =	
 LDFLAGS =	@LDFLAGS@
 LIBS =		@LIBS@
 NETLIBS =	@V_NETLIBS@
