From f12ba2a4e43e764fb0d352e5423e30de192dedef Mon Sep 17 00:00:00 2001
From: "Franklin S. Cooper Jr" <fcooperjr27@gmail.com>
Date: Fri, 7 Sep 2012 18:36:24 -0500

---
 extras/htpasswd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/extras/htpasswd.c b/extras/htpasswd.c
index 31ac769..dfe4c15 100644
--- a/extras/htpasswd.c
+++ b/extras/htpasswd.c
@@ -47,7 +47,7 @@ static void getword(char *word, char *line, char stop) {
     while((line[y++] = line[x++]));
 }
 
-static int my_getline(char *s, int n, FILE *f) {
+static int ht_getline(char *s, int n, FILE *f) {
     int i=0;
 
     while(1) {
@@ -188,7 +188,7 @@ int main(int argc, char *argv[]) {
     user[sizeof(user)-1] = '\0';
 
     found = 0;
-    while(!(my_getline(line,MAX_STRING_LEN,f))) {
+    while(!(ht_getline(line,MAX_STRING_LEN,f))) {
         if(found || (line[0] == '#') || (!line[0])) {
             putline(tfp,line);
             continue;
