From 49e49ca76d2c181772e74447b4f231ff260102d7 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Mon, 8 Aug 2011 16:59:16 -0400
Subject: [PATCH] opkg_message: send "Collected errors:" messages to stderr not stdout

Upstream-status: Submitted [http://code.google.com/p/opkg/issues/detail?id=63]

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 libopkg/opkg_message.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libopkg/opkg_message.c b/libopkg/opkg_message.c
index 7114e3a..2862ea6 100644
--- a/libopkg/opkg_message.c
+++ b/libopkg/opkg_message.c
@@ -64,10 +64,10 @@ print_error_list(void)
 	struct errlist *err = error_list_head;
 
 	if (err) {
-		printf("Collected errors:\n");
+		fprintf(stderr, "Collected errors:\n");
 		/* Here we print the errors collected and free the list */
 		while (err != NULL) {
-			printf(" * %s", err->errmsg);
+			fprintf(stderr, " * %s", err->errmsg);
 			err = err->next;
 		}
 	}
-- 
1.7.0.4

