From 00666002c48ebbb21e73eb9fbe90615cef87cc11 Mon Sep 17 00:00:00 2001
From: Denys Dmytriyenko <denys@ti.com>
Date: Mon, 14 Oct 2013 18:53:28 -0400
Subject: [PATCH] Update Makefile.build for when build dir is not the same as
 source dir

Signed-off-by: Denys Dmytriyenko <denys@ti.com>
---
 Makefile.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.build b/Makefile.build
index 18892d4..72b8e34 100644
--- a/Makefile.build
+++ b/Makefile.build
@@ -2,6 +2,7 @@
 
 ENV_SETUP ?= ../../linux-devkit/environment-setup
 DESTDIR ?= 
+BUILDDIR ?= .
 PLATFORM ?= 
 MATRIX_APP_DIR ?= /usr/share/matrix-gui-2.0/apps/
 PLATFORM_DEFINE ?= "Platform_${PLATFORM}"
@@ -28,7 +29,7 @@ clean : ${SOURCES} qmake
 
 install_common:
 	@install -d ${DESTDIR}/usr/bin 
-	@install -m 0755 ThermostatDemo ${DESTDIR}/usr/bin/ThermostatDemo
+	@install -m 0755 ${BUILDDIR}/ThermostatDemo ${DESTDIR}/usr/bin/ThermostatDemo
 	@install -d ${DESTDIR}${MATRIX_APP_DIR}/qt_tstat
 	@cp -rf matrix-files/*  ${DESTDIR}${MATRIX_APP_DIR}/qt_tstat
 
-- 
1.8.3.2

